app.py

```diff
@@ -56,7 +56,17 @@ def experience():
         The experience(s) as a JSON object, or an error message.
     '''
     if request.method == 'GET':
-        return jsonify(data.get('experience',[]))
+        index = request.args.get('index')
+        experiences = data.get('experience',[])
+        if index is not None:
+            try:
+                index = int(index)
+                if 0 <= index < len(experiences):
+                    return jsonify(experiences[index])
+                return jsonify({'error':'Index out of range'})
```

Comment:
It would be better here to return a 404 error with some generic description like "experience not found".

This is mainly from a security pov for the application as a malicious user could potentially guess the number of entries in the database (or storage) and could abuse it further.
Of course, this is not very relevant here given the usage of this code, but it's still good to keep in mind while working on this here.